cypress/e2e/pages/workAllocation.cy.ts

```diff
@@ -266,6 +266,91 @@ describe('Work Allocation', () => {
+    });
+    context('when a new project is been created successfully', () => {
+      before(() => {
+        cy.findByTestId('project').type('new project');
+        cy.findByRole('button', { name: /Save/i }).click();
+      });
+      it('hides the add new config option form', () => {
+        cy.findByLabelText('Add New Project').should('not.exist');
+        cy.findByRole('button', { name: /Save/i }).should('not.exist');
+      });
+      it('displays the project select box with the newly added project name', () => {
+        shouldDisplaySelectedValue('project', 'new project');
+      });
+    });
+    context('when a new project fails to be created', () => {
+      before(() => {
+        cy.msw().then(({ graphql, worker }) => {
+          worker.use(
+            graphql.mutation('AddProject', () => {
+              return HttpResponse.json({
+                data: {
+                  addProject: null
+                }
+              });
+            })
+          );
+        });
+        cy.findByTestId('addNewProject-btn').click();
+        cy.findByTestId('project').type('new project');
+        cy.findByRole('button', { name: /Save/i }).click();
+      });
+      it('keeps the add new config option form', () => {
+        cy.findByLabelText('Add New Project').should('be.visible');
+
+        cy.findByRole('button', { name: /Save/i }).should('be.visible');
+      });
+    });
+  });
+
+  context('when adding a new Cost code)', () => {
+    before(() => {
+      cy.visitAsEndUser('/sgp');
+      cy.findByTestId('addNewCostCode-btn').click();
+    });
+    context('when a new cost code is been created successfully', () => {
+      before(() => {
+        cy.findByTestId('costCode').type('S12345');
+        cy.findByRole('button', { name: /Save/i }).click();
+      });
+      it('hides the add new config option form', () => {
+        cy.findByLabelText('Add New Cost Code').should('not.exist');
+        cy.findByRole('button', { name: /Save/i }).should('not.exist');
+      });
+      it('displays the cost code select box with the newly added cost code', () => {
+        shouldDisplaySelectedValue('costCode', 'S12345');
+      });
+    });
+    context('when a new cost code fails to be created', () => {
+      before(() => {
+        cy.msw().then(({ graphql, worker }) => {
+          worker.use(
+            graphql.mutation('AddCostCode', () => {
+              return HttpResponse.json({
+                data: {
+                  addCostCode: null
+                }
+              });
+            })
+          );
+        });
+        cy.findByTestId('addNewCostCode-btn').click();
+        cy.findByTestId('costCode').type('S12345');
+        cy.findByRole('button', { name: /Save/i }).click();
+      });
+      it('keeps the add new config option form', () => {
+        cy.findByLabelText('Add New Cost Code').should('be.visible');
+        cy.findByRole('button', { name: /Save/i }).should('be.visible');
+      });
+    });
+  });
+
```

Comment:
Do we have a unit test cases for the input part of workAllocation page?  If so, these new test cases can be moved over there.

Reply:
the unit test case for the input are provided within the addNewConfigOption testing file, this is to test the interaction of the input file with the workAllocation component